e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js

```diff
@@ -0,0 +1,136 @@
+import cancelCancellationPage from '../../pages/deal-cancellation/cancel-cancellation';
+
+context('Deal cancellation - cancel cancellation', () => {
+  let dealId;
+  const dealFacilities = [];
+
+  before(() => {
+    cy.insertOneDeal(MOCK_DEAL_AIN, BANK1_MAKER1).then((insertedDeal) => {
+      dealId = insertedDeal._id;
+
+      const { dealType, mockFacilities } = MOCK_DEAL_AIN;
+
+      cy.createFacilities(dealId, [mockFacilities[0]], BANK1_MAKER1).then((createdFacilities) => {
+        dealFacilities.push(...createdFacilities);
+      });
+
+      cy.submitDeal(dealId, dealType, PIM_USER_1);
+    });
+  });
+
+  after(() => {
+    cy.deleteDeals(dealId, ADMIN);
+    dealFacilities.forEach((facility) => {
+      cy.deleteFacility(facility._id, BANK1_MAKER1);
+    });
+  });
+
+  describe('when logged in as a PIM user', () => {
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+    });
+
+    describe('when a cancellation is not in draft', () => {
+      beforeEach(() => {
+        cy.visit(relative(`/case/${dealId}/deal`));
+
+        caseDealPage.cancelDealButton().click();
+
+        cy.clickCancelLink();
+      });
+
+      it('should redirect to deal summary page', () => {
+        cy.url().should('eq', relative(`/case/${dealId}/deal`));
+      });
+    });
+
+    describe('when a cancellation is in draft', () => {
+      beforeEach(() => {
+        cy.visit(relative(`/case/${dealId}/deal`));
+
+        caseDealPage.cancelDealButton().click();
+
+        cy.keyboardInput(reasonForCancellingPage.reasonForCancellingTextBox().clear(), 'xxx');
+        cy.clickContinueButton();
+      });
+
+      describe('when visiting page by URL navigation', () => {
+        beforeEach(() => {
+          cy.visit(relative(`/case/${dealId}/cancellation/cancel`));
+        });
+
+        it('should render page correctly', () => {
+          cy.url().should('eq', relative(`/case/${dealId}/cancellation/cancel`));
+
+          backLink();
+          cancelCancellationPage.noGoBackButton();
+          cancelCancellationPage.yesCancelButton();
+        });
+
+        it(`back link navigates to reason page`, () => {
+          cy.clickBackLink();
+
+          cy.url().should('eq', relative(`/case/${dealId}/cancellation/reason`));
+        });
+
+        it(`no, go back button navigates to reason page`, () => {
+          cancelCancellationPage.noGoBackButton().click();
+
+          cy.url().should('eq', relative(`/case/${dealId}/cancellation/reason`));
+        });
+
+        it('yes cancel button navigates to deal summary page', () => {
+          cancelCancellationPage.yesCancelButton().click();
+
+          cy.url().should('eq', relative(`/case/${dealId}/deal`));
+        });
+
+        it('yes button wipes the cancellation data', () => {
+          cancelCancellationPage.yesCancelButton().click();
+
+          caseDealPage.cancelDealButton().click();
+          reasonForCancellingPage.reasonForCancellingTextBox().should('have.value', '');
+        });
```

Comment:
Does the deal status change after cancellation? If so, would be great to check the status hasn't changed after cancelling the cancellation

Reply:
Not yet - this is a future PR